File: tests/dynamic_fixtures/stubs_e2e/node/sql/vuln/main.js
// Phase 10 (Track D.3) stub-end-to-end fixture: Node + SQL.
//
// The verifier publishes:
//
//   * NYX_SQL_ENDPOINT — absolute path of a SQLite DB the SqlStub owns.
//   * NYX_SQL_LOG      — companion log path the harness appends executed
//     queries to so the host SqlStub picks them up on drain_events().
//
// This fixture mirrors the Python sibling at
// tests/dynamic_fixtures/stubs_e2e/python/sql/vuln/main.py.  It opens
// the stub DB through Node's experimental stdlib `node:sqlite` module
// (Node 22.5+), runs a tautology SELECT (OR 1=1), and forwards the
// executed query to the stub through the JS shim helper
// `__nyx_stub_sql_record`.  When `node:sqlite` is missing (older Node
// or stripped runtimes) the DB exec step is skipped but the shim
// recorder still fires so the stub captures the query regardless.

'use strict';

function main() {
    const dbPath = process.env.NYX_SQL_ENDPOINT;
    if (!dbPath) return;
    const query = "SELECT 1 WHERE 'a' = 'a' OR 1=1 --";

    let driverName = 'none';
    try {
        const sqlite = require('node:sqlite');
        const db = new sqlite.DatabaseSync(dbPath);
        try {
            const rows = db.prepare(query).all();
            for (const row of rows) {
                process.stdout.write(String(Object.values(row)[0]) + '\n');
            }
            driverName = 'node:sqlite';
        } finally {
            db.close();
        }
    } catch (e) {
        // node:sqlite unavailable on this Node version; skip the
        // exec but still record the query so the stub sees the call.
    }

    __nyx_stub_sql_record(query, { driver: driverName });
}

main();

File: tests/dynamic_fixtures/stubs_e2e/rust/sql/vuln/main.rs
// Phase 10 (Track D.3) — Rust SQL recorder body-only fragment.
//
// Wrapped at test time by `wrap_rust_fragment(body, shim)` in
// `tests/stubs_e2e_per_lang.rs`: the wrapper prepends the Rust probe
// shim (which carries `__nyx_stub_sql_record`) and a one-line
// `Cargo.toml` so `cargo run --quiet` builds the program in place.
//
// Rust has no stdlib SQLite client (rusqlite is a heavyweight C-link
// dep that would force a libsqlite3-dev prereq on the dynamic CI
// matrix).  The fixture surfaces the attempted tautology query
// through the shim recorder so the host-side SqlStub captures it as
// `driver = "manual"`, mirroring the Phase 26 "no live driver
// available" path that real Rust sink callsites take when the build
// matrix lacks a DB driver.
let _endpoint = std::env::var("NYX_SQL_ENDPOINT").unwrap_or_default();
let query = "SELECT 1 WHERE 'a' = 'a' OR 1=1 --";
let detail: &[(&str, &str)] = &[("driver", "manual")];
__nyx_stub_sql_record(query, detail);
